### ISSUE TITLE
Add a TypeScript definition

### ISSUE THREAD

**Issue by aendrew** [michael/github#241]
This will help adoption by people who use TypeScript and doesn't impact the library any other way.

I'll do this because I want it and have experience with TS.

**Comment by aendrew** [comment]
Anyone interested, I've started work at aendrew/github#typescript_def. It's completely unfinished and I'm struggling with the syntax, so any help is quite welcome.

**Comment by aendrew** [comment]
Apparently you can generate these via jsDoc. I'd really prefer to write these all in one place and then have tooling (Possibly [this](https://github.com/fivetran/typescript-closure-tools)) generate any copies, so will add these to the definition for the moment and then migrate that documentation into the actual code itself sometime around 0.11.0 (I'd do so now but it'd change the shape of the code enough that it'd likely cause a large number of existent PRs to conflict).

**Comment by AurelioDeRosa** [comment]
@aendrew Are you still interested in this?

**Comment by aendrew** [comment]
@AurelioDeRosa Yep, absolutely, just a bit delayed. Probably a good thing though; ambient definitions have changed a bit since I posted this. Currently researching the best way to add a def to a lib.

**Comment by clayreimann** [comment]
@aendrew We're all jsdoc'd up so integrating this shouldn't be too hard if you want to.

**Comment by smakinson** [comment]
@aendrew Have you made any progress on your research?

**Comment by aendrew** [comment]
Here's a first pass, I have no idea if it actually works yet — I've merely generated the d.ts from jsDoc and haven't had time to play with the definition beyond that: https://github.com/aendrew/github/tree/d.ts

**Comment by aendrew** [comment]
See above PR if you want to test the definition I've been working on. Automatic generation is about 95% of the way there; once tsd-jsdoc is fixed upstream it'll be worth adding d.ts generation as a build step.
